eox_nelp/signals/tests/test_receivers.py

```diff
@@ -481,3 +486,58 @@ def test_event_is_emitted(self, tracker_mock):
                 "created": block.created,
             }
         )
+
+
+class IncludeTrackerContextTestCase(unittest.TestCase):
+    """Test class for include_tracker_context method."""
+
+    def test_context_is_included(self):
+        """
+        This tests that the body kwargs has been updated after the method execution.
+
+        Expected behavior:
+            - body kwargs is equal to the tracker context
+        """
+        body = {"kwargs": {}}  # This is the default value for kwargs
+        context = {"This is a fake context": True}
+
+        # Set tracker context
+        tracker = get_tracker()
+
+        with tracker.context("this_does_not_matter", context):
+            include_tracker_context(body)
+
+        self.assertEqual(context, body["kwargs"]["tracker_context"])
+
+
+class UpdateAsyncTrackerContextTestCase(unittest.TestCase):
+    """Test class for update_async_tracker_context method."""
+
+    def test_context_is_included(self):
+        """
+        This tests that the tracker context has been updated after the method execution.
+
+        Expected behavior:
+            - sender request doesn't have any tracker_context key.
+            - the current tracker returns the expected context.
+        """
+        sender = Mock()
+        expected_context = {
+            "session_id": "ashjdgjashgdui15647851561",
+            "course_id": "course-v1:test+Cx105+2022_T4"
+        }
+        sender.request = {
+            "kwargs": {
+                "tracker_context": expected_context
+            }
+        }
+        tracker = get_tracker()
+
+        update_async_tracker_context(sender)
+
+        self.assertEqual(expected_context, tracker.resolve_context())
+        self.assertNotIn("tracker_context", sender.request["kwargs"])
+
+        # Following line doesn't affect the current implementation, this
+        # just clean the context affected by this test.
```

Comment:
This could be in tearDown section?

Reply:
that was my first approach however that required a more complex implementation so I decided to keep this way since I just needed one test case